lib/rubocop/cop/rspec/predicate.rb

```diff
@@ -0,0 +1,182 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Prefer using predicate matcher instead of truthiness
+      #
+      # @example
+      #   # bad
+      #     expect(foo.valid?).to be_truthy
+      #
+      #   # good
+      #     expect(foo).to be_valid
+      class Predicate < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Use predicate matcher `%s`'.freeze
+
+        def on_send(node)
+          expectation = correction(node)
+          return if expectation.nil?
+
+          add_offense(
+            node,
+            :expression,
+            format(MSG, expectation)
+          )
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(node.loc.expression, correction(node))
+          end
+        end
+
+        private
+
+        def requires_predicate?
+          style == :predicate
+        end
+
+        def correction(node)
+          if requires_predicate?
+            PredicateCorrector.new(node, style).correction
+          else
+            BooleanCorrector.new(node, style).correction
+          end
+        end
+      end
+
+      # Corrector base class for offense detection
+      class Corrector
+        attr_reader :node, :style, :object, :args, :action
+
+        def initialize(node, style)
+          @node = node
+          @style = style
+
+          receiver, @action, @args = *node
+          _, _, @object = *receiver
+        end
+
+        def correction
+          return unless correctable?
+
+          node.source
+            .gsub(object.source, expectation)
+            .gsub(args.source, matcher)
+            .gsub(".#{action}", actionable)
+        end
+
+        def correctable?
+          return false unless object
+          return false unless args
+          return false if args.source =~ /^(be_truthy|be_falsey)$/
+          return false unless expectation
+          return false unless matcher
+
+          true
+        end
+      end
+
+      # Corrector for the predicate style
+      class PredicateCorrector < Corrector
+        private
+
+        def actionable
+          if args.source.include?('false')
+            '.not_to'
```

Comment:
Sure. This is just my observation as a user of the auto correct feature by the way. I'm fairly certain this is the design but just be aware that in this case I'm not familiar with the actual implementation details